src/Content/Conversation/Factory/Timeline.php

```diff
@@ -108,13 +135,21 @@ public function getNetworkFeeds(string $command): Timelines
 		return new Timelines($tabs);
 	}
 
+	public function isNetwork(string $selectedTab): bool
```

Comment:
This is a good example of what I wanted you to introduce earlier with Entity inheritance. It looks like there are four types of Timelines: NetworkTimeline, CommunityTimeline and ChannelTimeline. Under ChannelTimeline, we can find UserCreatedChannelTimelines as well.

This for me creates a neat entity inheritance tree:
```
           __________Timeline______________
         /              |                  \
NetworkTimeline CommunityTimeline     ChannelTimeline
                                            |
                                  UserCreateChannelTimeline
```
This way you can directly interrogate the type of an Entity to know if it's a Network timeline, a Community timeline, a Channel timeline or a user-created Channel timeline. Then you don't need these static-looking methods in this class.

Then you can tweak `getCommunities()`, `getNetworkFeeds()` and `getChannelsForUser` to return a collection of the corresponding entity.

Reply:
I guess I mostly understand, what you mean. But I completely don't know, how to achieve this.